File: app/src/main/java/com/example/olesya/boardgames/ItemTouchCallback.java
package com.example.olesya.boardgames;

import android.support.v7.widget.RecyclerView;
import android.support.v7.widget.helper.ItemTouchHelper;

import com.example.olesya.boardgames.adapter.CardPagerAdapter;

public class ItemTouchCallback extends ItemTouchHelper.Callback {

    private CardPagerAdapter mAdapter;

    public ItemTouchCallback(CardPagerAdapter adapter) {
        mAdapter = adapter;
    }

    private boolean isSwipeEnabled = false;

    @Override
    public boolean isItemViewSwipeEnabled() {
        return isSwipeEnabled;
    }

    @Override
    public int getMovementFlags(RecyclerView recyclerView, RecyclerView.ViewHolder viewHolder) {
        int dragFlags = ItemTouchHelper.START | ItemTouchHelper.END;
        int swipeFlags = ItemTouchHelper.UP;
        return makeMovementFlags(dragFlags, swipeFlags);
    }

    @Override
    public boolean onMove(RecyclerView recyclerView, RecyclerView.ViewHolder viewHolder, RecyclerView.ViewHolder target) {
        return false;
    }

    @Override
    public void onSwiped(RecyclerView.ViewHolder viewHolder, int direction) {
//        mAdapter.onRemove(viewHolder.getAdapterPosition());
    }

    public void setSwipeEnabled(boolean isSwipeEnabled) {
        this.isSwipeEnabled = isSwipeEnabled;
    }

}

File: app/src/main/java/com/example/olesya/boardgames/ui/ItemTouchCallback.kt
package com.example.olesya.boardgames.ui

import androidx.lifecycle.MutableLiveData
import androidx.recyclerview.widget.ItemTouchHelper
import androidx.recyclerview.widget.RecyclerView

class ItemTouchCallback : ItemTouchHelper.Callback() {

    var pickedData: MutableLiveData<Int> = MutableLiveData()

    var isSwipeEnabled = false

    override fun getMovementFlags(recyclerView: RecyclerView, viewHolder: RecyclerView.ViewHolder): Int {
        return ItemTouchHelper.Callback.makeMovementFlags(ItemTouchHelper.START or ItemTouchHelper.END,
                if (isSwipeEnabled) ItemTouchHelper.UP else 0)
    }

    override fun onMove(recyclerView: RecyclerView, viewHolder: RecyclerView.ViewHolder, target: RecyclerView.ViewHolder): Boolean {
        return false
    }

    override fun onSwiped(viewHolder: RecyclerView.ViewHolder, direction: Int) {
        val position = viewHolder?.adapterPosition ?: return
        pickedData.postValue(position)
    }
}